src/memkind_memtier.c

```diff
@@ -1189,14 +1192,63 @@ MEMKIND_EXPORT void *memtier_malloc(struct memtier_memory *memory, size_t size)
     void *ptr;
     uint64_t data;
 
-    ptr = memtier_kind_malloc(memory->get_kind(memory, size, &data), size);
+    dont_mmap = true;
+    memkind_t kind = memory->get_kind(memory, size, &data);
+    ptr = memtier_kind_malloc(kind, size);
     memory->post_alloc(data, ptr, size);
     memory->update_cfg(memory);
     print_memory_statistics(memory);
 
+    dont_mmap = false;
     return ptr;
 }
 
+void *sys_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t off)
+{
+    long ret = syscall(SYS_mmap, addr, length, prot, flags, fd, off);
+    if (ret == -EPERM && !off && (flags&MAP_ANON) && !(flags&MAP_FIXED))
+        ret = -ENOMEM;
+    if (ret > -4096 && ret < 0) {
+        errno = -ret;
+        return MAP_FAILED;
+    }
+
+    return (void*)ret;
+}
+
+int munmap(void *addr, size_t length)
```

Comment:
How does this implementation differ from the original munmap? Is it necessary to add it here?